corehq/apps/app_manager/views/utils.py

```diff
@@ -429,7 +429,7 @@ def _get_mm_map_by_id(multimedia_map):
     target_build = get_app_cached(domain, target_build_id)
     assert source_build.copy_of, _("Size calculation available only for builds")
     assert target_build.copy_of, _("Size calculation available only for builds")
-    build_profile = source_build.build_profiles[build_profile_id] if build_profile_id else None
+    build_profile = source_build.build_profiles.get(build_profile_id) if build_profile_id else None
```

Comment:
It'd be better to deal with this by eliminating the invalid options from the dropdown in the first place - just pass the intersection of app A's profiles and app B's profiles, instead of passing all of app A's profiles.

Reply:
hmm, how about cases where you are updating from a profile that is present in one but not in other. I think that is a possible case.

Reply:
I think keeping this code is fine for the sake of being robust to future changes, but also still think it would be good to limit what shows in the dropdown by updating the `build_profiles` added to `corehq/apps/app_manager/views/releases.py`.

Reply:
Not clear by what you mean by "build profiles added", You mean manual changes to the build profiles and doing something fishy?

Reply:
I mean changing this code: https://github.com/dimagi/commcare-hq/blob/d6e680fff568aa283e813a49297751119cbc364b/corehq/apps/app_manager/views/releases.py#L572 to take the intersection of `self.first_app.build_profiles` and `self.second_app.build_profiles`

(Sorry this comment thread has been confusing - I should have commented on that code in the first place, I was in a rush when re-commenting and had trouble finding it. And now that we're talking here, I'm just going to keep commenting on this thread.)

Reply:
Just to share the context and as to why i am leaning towards this,

On the Source files page, the first app is the app we are currently looking at and the other app is the version we are comparing to.
In the view where I have added comparison, it says "Updating from other app version to this app version", assuming the other app would be the lower version.
So in case of an app update done on phone,
the edge case i am thinking about is updating from an older version where i had installed a specific build profile which is not available anymore and hence the update would be to default now.
For cases if a specific build profile is newly introduced, we can't update to that anyway, it has to be only freshly installed.
So on those lines this should be the build profiles of the other app or the app with lower version. Taking intersection would ignore the build profiles that have been removed after the version that is present on the phone.

which means i should change this to `self.other_app` and not `self.first_app`.

Reply:
offlined to collect feedback on this post deploy and then see if and how we should be changing it